numalogic/udfs/postprocess.py

```diff
@@ -0,0 +1,186 @@
+
+        Args:
+            config_id: Config ID
+            conf: StreamConf object
+        """
+        self.pl_conf.stream_confs[config_id] = conf
+
+    def get_conf(self, config_id: str) -> StreamConf:
+        try:
+            return self.pl_conf.stream_confs[config_id]
+        except KeyError as err:
+            raise ConfigNotFoundError(f"Config with ID {config_id} not found!") from err
+
+    def exec(self, keys: list[str], datum: Datum) -> Messages:
+        """
+        The postprocess function here receives data from the previous udf.
+
+        Args:
+        -------
+        keys: List of keys
+        datum: Datum object.
+
+        Returns
+        -------
+        Messages instance
+
+        """
+        _start_time = time.perf_counter()
+        messages = Messages()
+
+        # Construct payload object
+        payload = StreamPayload(**orjson.loads(datum.value))
+
+        # load configs
+        thresh_cfg = self.get_conf(payload.config_id).numalogic_conf.threshold
+        postprocess_cfg = self.get_conf(payload.config_id).numalogic_conf.postprocess
+
+        # load artifact
+        thresh_artifact = _load_model(
+            skeys=keys, dkeys=[thresh_cfg.name], payload=payload, model_registry=self.model_registry
+        )
+        postproc_clf = self.postproc_factory.get_instance(postprocess_cfg)
+
+        if thresh_artifact is None:
+            payload = replace(
+                payload, status=Status.ARTIFACT_NOT_FOUND, header=Header.TRAIN_REQUEST
+            )
+
+        #  Postprocess payload
+        if payload.status in (Status.ARTIFACT_FOUND, Status.ARTIFACT_STALE) and thresh_artifact:
+            try:
+                anomaly_scores = self.compute(
+                    model=thresh_artifact.artifact,
+                    input_=payload.get_data(),
+                    postproc_clf=postproc_clf,
+                )
+            except RuntimeError:
+                _LOGGER.exception(
+                    "%s - Runtime postprocess error! Keys: %s, Metric: %s",
+                    payload.uuid,
+                    payload.composite_keys,
+                    payload.metrics,
+                )
+                payload = replace(payload, status=Status.RUNTIME_ERROR, header=Header.TRAIN_REQUEST)
+            else:
+                payload = replace(
+                    payload,
+                    data=anomaly_scores,
+                    header=Header.MODEL_INFERENCE,
+                )
+                out_payload = OutputPayload(
+                    uuid=payload.uuid,
+                    config_id=payload.config_id,
+                    composite_keys=payload.composite_keys,
+                    timestamp=payload.end_ts,
+                    unified_anomaly=np.max(anomaly_scores),
+                    data={
+                        _metric: _score for _metric, _score in zip(payload.metrics, anomaly_scores)
+                    },
+                    metadata=payload.metadata,
```

Comment:
we need to add model versions, std, mean and isAnomaly

Reply:
what is std and mean?

Reply:
std is standard deviation, mean is mean score I think @s0nicboOm

Reply:
That I understand, but std of what?

Reply:
could you look into current OSAM what they are publishing?

Reply:
it is threshold model’s mean and std

Reply:
@nkoppisetty Do we have the above fields in publisher code? If yes, will need to know the datatypes for them.
In old OSAM they were a different field. I feel they should go as a metadata here for flexibility and open source.

Reply:
Adding a TODO. Ideally we should emit as UDF metrics.